File: LinkedList/delete_node.cpp
#include <iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
};
void delete_node(Node* head,int index){
    Node* temp=head;
    int count=0;
    while(temp->next!=NULL){
        if(count!=index){
            cout<<temp->data<<" ";
        }
        count++;
        temp=temp->next;
    }
    if(count!=index){
        cout<<temp->data;
    }
}
int main(int argc,const char *argv[]){
    int index=atoi(argv[1]);
    int n=argc-2;
    Node* node[n];
    for(int i=0;i<n;i++){
        node[i]=NULL;
        node[i]=new Node;
    }
    int j=0;
    for(int i=2;i<argc;i++){
        node[j]->data=atoi(argv[i]);
        node[j]->next=node[j+1];
        j++;
    }
    node[n-1]->data=atoi(argv[argc-1]);
    node[n-1]->next=NULL;
    Node* head=node[0];
    delete_node(head,index);
    return 0;
}

File: LinkedList/delete_node.java
import java.io.*;
import java.util.Scanner; 
class Node {
    int data;
    Node next;
    Node(int d) {
        this.data = d;
        this.next = null;
    }
}
 class LinkedList{
    public static Node cur2=null;
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t-->0){
            Node head=null;
            Node cur=null;
        int n=sc.nextInt();
        int p=sc.nextInt();
        cur=null;
        for(int i=0;i<n;i++)
        {
            int d=sc.nextInt();
            Node ptr=new Node(d);
            if(head==null)
            {
                head=ptr;
                cur=ptr;
            }
            else
            {
                cur.next=ptr;
                cur=ptr;
            }
        }
        cur2=head;
        while(p-->0)
        {
            cur2=cur2.next;
        }
        GfG g=new GfG();
        g.deleteNode(cur2);
        while(head!=null)
        {
            System.out.print(head.data+" ");
            head=head.next;
        }
        System.out.println();
        }
    }
}




class GfG{
    void deleteNode(Node node) {
        Node temp=node;
        temp = temp.next;
        node.data=temp.data;
        node.next=temp.next;
    }
}
